moonraker/components/notifier.py

```diff
@@ -0,0 +1,133 @@
+# Moonraker Notifier
+#
+# Copyright (C) 2022 Pataar <me@pataar.nl>
+#
+# This file may be distributed under the terms of the GNU GPLv3 license.
+
+from __future__ import annotations
+
+import apprise
+import logging
+
+# Annotation imports
+from typing import (
+    TYPE_CHECKING,
+    Type,
+    Optional,
+    Dict,
+    Any,
+    List,
+)
+
+if TYPE_CHECKING:
+    from confighelper import ConfigHelper
+    from . import klippy_apis
+
+    APIComp = klippy_apis.KlippyAPI
+
+
+class Notifier:
+    def __init__(self, config: ConfigHelper) -> None:
+        self.server = config.get_server()
+        self.notifiers: Dict[str, NotifierInstance] = {}
+        self.events: Dict[str, NotifierEvent] = {}
+        prefix_sections = config.get_prefix_sections("notifier")
+
+        self.register_events(config)
+
+        for section in prefix_sections:
+            cfg = config[section]
+            notifier_class: Optional[Type[NotifierInstance]]
+            try:
+                notifier = NotifierInstance(cfg)
+
+                for event in self.events:
+                    if event in notifier.events or "*" in notifier.events:
+                        self.events[event].register_notifier(notifier)
+
+                logging.info(f"Registered notifier: '{notifier.get_name()}'")
+
+            except Exception as e:
+                msg = f"Failed to load notifier[{cfg.get_name()}]\n{e}"
+                self.server.add_warning(msg)
+                continue
+            self.notifiers[notifier.get_name()] = notifier
+
+    def register_events(self, config: ConfigHelper):
```

Comment:
I'm ok with hardcoding events to start, but we might want to try to think of a way of allowing users to define events.  This is something we can explore later though.

Reply:
I agree though